libsolutil/CommonData.h

```diff
@@ -554,7 +554,7 @@ std::string formatAsStringOrNumber(std::string const& _value);
 
 /// @returns a string with the usual backslash-escapes for non-ASCII
 /// characters and surrounded by '"'-characters.
-std::string escapeAndQuoteString(std::string const& _input);
+std::string escapeAndQuoteYulString(std::string const& _input);
```

Comment:
This function is currently only used for Yul strings so I just renamed it instead of adding a parameter for selecting between languages.

If you think we should instead make this function handle both languages, see the first commit without the fixup. I can drop the fixup if that's better.

Reply:
But it has nothing to do with Yul though.

Reply:
It didn't but now it does because sets of escapes supported by Soldity and Yul are different. `escapeAndQuoteYulString()` as it is now only uses escapes allowed in Yul.